apps/crawler/crawler/s3/crawler/s3_crawler.py

```diff
@@ -84,30 +105,44 @@ def _is_new_object(self, object_metadata: dict[str, Any], file_path: str) -> boo
             # TODO: parth - if I change this to return False, no test fails
             return True
 
-    def _get_s3_client(
-        self, boto_session_args: list[Any], boto_session_kwargs: dict[str, Any]
-    ) -> botocore.client.BaseClient:
-        session = boto3.session.Session(*boto_session_args, **boto_session_kwargs)
-        return session.client("s3")
+    def _get_s3_client(self) -> botocore.client.BaseClient:
+        if self._anon:
+            cfg = Config(signature_version=UNSIGNED)
+            return self._session.client("s3", config=cfg)
+        else:
+            return self._session.client("s3")
 
 
 if __name__ == "__main__":
     print("Version-Info, Sycamore Crawler S3 Branch:", os.environ.get("GIT_BRANCH", "unset"))
     print("Version-Info, Sycamore Crawler S3 Commit:", os.environ.get("GIT_COMMIT", "unset"))
     print("Version-Info, Sycamore Crawler S3 Diff:", os.environ.get("GIT_DIFF", "unset"))
 
-    if len(sys.argv) > 3 or (len(sys.argv) > 1 and sys.argv[1] == "-h"):
-        print("Usage : poetry run python s3_crawler.py bucket_name prefix_value")
-    else:
-        if len(sys.argv) == 1:
-            bucket = "aryn-public"
-            prefix = "sort-benchmark"
-        elif len(sys.argv) == 2:
-            bucket = sys.argv[1]
-            prefix = ""
+    anon = False
```

Comment:
Lets use argparse. It's part of the standard library.  Rather than rolling our own (even if it's simple)
https://docs.python.org/3/library/argparse.html

Reply:
argparse it is.  Click looked like a more extensive job.